### PR TITLE
Feat/viewmodel coroutine sample

### PR DESCRIPTION
**Pull request by gorbuvla**
Sample app using viewModels and coroutine scopes \+ channels

### CODE REVIEW

File: coroutineoauth/src/main/java/cz/ackee/ackroutine/core/OAuthInterceptor.kt [hunk 1]
```diff
@@ -15,7 +16,10 @@ class OAuthInterceptor internal constructor(private val oAuthStore: OAuthStore)
 
         if (!TextUtils.isEmpty(oAuthStore.accessToken)) {
             val accToken = oAuthStore.accessToken
+            Log.d("OKHTTP", "Setting token header $accToken")
             builder.addHeader("Authorization", "Bearer $accToken")
+        } else {
+            Log.d("OKHTTP", "No access token")
```
Comment (gorbuvla) [inline coroutineoauth/src/main/java/cz/ackee/ackroutine/core/OAuthInterceptor.kt]: remove debug logs

File: retrofitadapter/src/main/java/cz/ackee/retrofitadapter/chain/CallChain.kt [hunk 2]
```diff
@@ -8,24 +8,46 @@ interface CallChain {
...
+//    private val annotationArray: Array<out Annotation>,
+//    private val interceptors: List<CallFactoryInterceptor>
+//) {
+//    fun proceed(call: Call<*>): Deferred<*> {
+//        if (chainIndex + 1 > interceptors.size) {
+//            throw IllegalStateException("chainIndex ${chainIndex + 1} does not match with actual interceptor size ${interceptors.size}")
+//        }
+//
+//        val newChain = CallChainE(chainIndex + 1, call, annotationArray, interceptors)
+//        return interceptors[chainIndex].intercept(newChain)
```
Comment (gorbuvla) [inline retrofitadapter/src/main/java/cz/ackee/retrofitadapter/chain/CallChain.kt]: remove commented code

File: retrofitadapter/src/main/java/cz/ackee/retrofitadapter/AckroutineCallAdapterFactory.kt [hunk 3]
```diff
@@ -39,6 +40,6 @@ class AckroutineCallAdapterFactory(vararg interceptor: CallFactoryInterceptor) :
             BodyCallExecuteInterceptor()
         }
 
-        return AckroutineCallAdapter<Any>(responseType, interceptors + callInterceptor)
+        return AckroutineCallAdapter<Any>(responseType, annotations,interceptors + callInterceptor)
```
Comment (davidbilik) [inline retrofitadapter/src/main/java/cz/ackee/retrofitadapter/AckroutineCallAdapterFactory.kt]: weirdly placed comma, it seems that space is missing after that 🕵️

**Comment by davidbilik** [comment]
Good for me 👍

File: retrofitadapter/src/main/java/cz/ackee/retrofitadapter/AckroutineCallAdapterFactory.kt (hunk 3 above)
Comment (gorbuvla) [reply retrofitadapter/src/main/java/cz/ackee/retrofitadapter/AckroutineCallAdapterFactory.kt]: ✅

**Pull request merged**